overlord/ifacestate/ifacemgr.go

```diff
@@ -275,9 +275,23 @@ func (m *InterfaceManager) doRemoveSnapSecurity(task *state.Task, _ *tomb.Tomb)
 	if err != nil {
 		return err
 	}
-	// TODO: remove all connections from the state
-	if err := m.repo.RemoveSnap(snapName); err != nil {
 
+	conns, err := getConns(task.State())
```

Comment:
Then, on snapstate.Remove, distinguish between last and old, and conditionally inject discard-conns after remove-profiles.